confluentinc/ksql
# feat: CommandRunner enters degraded states when it processes command with higher version than it supports

### Description 
Stacks on top of 
https://github.com/confluentinc/ksql/pull/6012
https://github.com/confluentinc/ksql/pull/6031

Review only last commit
Prevents the server from processing a command from a newer version than it. I've repurposed the Version flag added https://github.com/confluentinc/ksql/pull/5104 as it wouldn't find any use currently since we can't break backwards compatibility currently.

### Testing done 
Unit test
Manual test:

- Start with fresh command topic
- Start up server on this branch
- Issue a bunch of statements to the command topic (CS/CSAS)
- Stopped the server
- Created a new branch and increment the VERSION in Command.java which is the ExpectedVersion
- Start server and issue DDL statements
- Switched to this branch and start up server
- Verify with JConsole the CommandRunnerStatus is degraded
- Verify all previous commands before the incompatible command are executed and the queries are started
- Verify CommandRunner thread is shut down by trying to issue more DDL statements to server
```
create stream steven as select * from test;
DDL command failed to be sent to command topic, the CommandRunner is in a DEGRADED state currently.
This can happen when it encounters an incompatible command when processing the command topic.
Check the server logs for more details.
```

### Reviewer checklist
